packages/react-jss/test-utils/createDynamicStylesTests.js

```diff
@@ -321,9 +321,9 @@ export default ({createStyledComponent}) => {
             color: red;
           }
         }
-          .container-d2-2 {  }
```

Comment:
it broke the order it seems, otherwise why is this diff here?

Reply:
if you don't konw the reason, I guess I need to go deep to find out the reason

Reply:
yeah, can't tell just by looking at this changes

Reply:
I found the reason, it because React-jss use `sheet.addRule` to handle dynamic rules, in this case, it will add:
```
container: {
          display: 'block',
          '@media (min-width: 0px)': {
            color: ({color}) => color
          }
        }
```
because it directly using `addRule` so container will be added as `container-d0`, then jss will processStyle which:
```
{
          display: 'block',
          '@media (min-width: 0px)': {
            color: ({color}) => color
          }
        }
```
then using `replaceRule` to `@media (min-width: 0px)` since it is a `NestedConditional` rule, so it won't increase the `counter` of RuleList. 

In the code before, we use `addRule` in nest plugin, so the dynamic `@media (min-width: 0px)` will increase the `counter`.

Now I'm wondering if it is safe to use `replaceRule` for all case, even if I can pass all the test cases.

Reply:
Where do you see the danger here? I might not be understanding the problem correctly.

Reply:
```
let key = name
    if (!isReplace && name in this.raw) {
      key = `${name}-d${this.counter++}`
    }

    this.raw[key] = decl
```
in this case, if we use old code, `this.raw` will be:
```
{
  container: {},
  @media (min-width: 0px): {},
  container-d0: {},
  @media (min-width: 0px)-d1: {},
}
```
the new add rule won't replace old rule. 

The main problem is, if people use `addRule` to add a nest rule, then it will use `replaceRule` inside. I think if people want to add rule, it should be really `add` but not `replace`.

Reply:
Can you please add a test that reproduces the "wrong" behavior in react-jss? So we can think of better ways of fixing it.